src/dnx.common/utils.cpp

```diff
@@ -83,5 +83,19 @@ namespace dnx
             return attributes != INVALID_FILE_ATTRIBUTES && ((attributes & FILE_ATTRIBUTE_DIRECTORY) == 0);
         }
 #endif
+
+        xstring_t remove_file_from_path(const xstring_t& path)
+        {
+            if (path.back() == _X('\\') || path.back() == _X('/'))
```

Comment:
It does not really matter here. But since we already have two checks like this I just created the `ends_with_slash` function where I use a variable. One issue I have not noticed before was that calling `.back()` on an empty string causes UB. In the function I created I check that the string is not empty before calling `.back()`